src/theory/strings/rewrites

```diff
@@ -285,11 +320,16 @@
   (str.<= s "")
   (= s ""))
 
-(define-cond-rule str-leq-concat ((s String :list) (t1 String) (s1 String) (t2 String :list) (s2 String :list))
+(define-cond-rule str-leq-concat-false ((s String :list) (t1 String) (s1 String) (t2 String :list) (s2 String :list))
   (and (= (str.len t1) (str.len s1)) (= (str.<= t1 s1) false))
   (str.<= (str.++ s t1 t2) (str.++ s s1 s2))
   false)
 
+(define-cond-rule str-leq-concat-true ((s String :list) (t1 String) (s1 String) (t2 String :list) (s2 String :list))
```

Comment:
In this case it doesn't matter since "true" occurs on the right hand side. 
The RARE reconstruction algorithm does take this into account, e.g. if we are trying to prove `t = true` and `t = t'` is proven, then it will try `t' = true` via TRANS.